### PR TITLE
Minor optimizations for TS put requests

### PR DESCRIPTION
**Pull request by macintux**
- Don't embed the local key in the Riak object when it will be
  immediately extracted and removed
- Directly pass the bucket, keys, and encoding function from PB put
  rather than intuiting them from the Riak object

### CODE REVIEW

File: src/riak_kv_w1c_worker.erl [hunk 1]
```diff
@@ -337,6 +333,22 @@ get_request_record(ReqId, #state{entries=Entries} = _State) ->
...
+%% Utility function for put/2: is this a TS object with its special
+%% requirements or a more traditional KV object?
+%%
+%% When riak_kv_pb_timeseries is driving a put request, it can provide
+%% all of these details directly to async_put/5, but when a tombstone
+%% is being put via riak_kv_delete, these must be extracted from the
+%% object.
+kv_or_ts_details(RObj, {ok, LocalKey}) ->
+    MD  = riak_object:get_metadata(RObj),
+    MD1 = dict:erase(?MD_TS_LOCAL_KEY, MD),
```
Comment (hmmr) [inline src/riak_kv_w1c_worker.erl:345]: There will be no local key to erase here, as you are no longer adding it in the metadata in the first plase (per changes in `riak_kv_timeseries:put_data`). Or am I missing something?

Comment (macintux) [inline src/riak_kv_w1c_worker.erl:345]: This is for scenarios where `riak_kv_w1c_worker:put` is invoked by an alternate path (specifically, putting a tombstone during a delete request).

Comment (hmmr) [inline src/riak_kv_w1c_worker.erl:345]: Ok. I was going to argue that we wouldn't need local key in metadata for deletes either, but then remembered that we in fact will.

I'll give my +1 as long as ts_basic passes on it (which I understand it does).

**Comment by hazen** [comment]
```
Test Results:
ts_basic-riak_kv_bitcask_backend,riak_kv_eleveldb_backend: pass
---------------------------------------------
0 Tests Failed
1 Tests Passed
That's 100.0% for those keeping score
```

**Comment by hazen** [comment]
:+1:  b5a42f9

**Comment by macintux** [comment]
@borshop merge

**Pull request merged**